Doc: update documentation of input parameters for four parts (#2530)
deepmodeling/abacus-develop: docs/advanced/input_files/input-main.md

### stru_file

- **Type**: String

> - **Description**: Specifies the name of the structure file containing various information about atom species, including pseudopotential files, local orbitals files, cell information, atom positions, and whether atoms should be allowed to move.

Reviewer: ```suggestion
- **Description**: The name of the structure file containing various information about atom species, including pseudopotential files, local orbitals files, cell information, atom positions, and whether atoms should be allowed to move.
```

- **Default**: STRU

### kpoint_file

- **Type**: String

> - **Description**: Specifies the name of the k-points file. Note that if you use atomic orbitals as basis and only use the gamma point, you don't need a k-point file in your directory. ABACUS will automatically generate a `KPT` file. However, if you use more than one k-point, please remember that the algorithm in ABACUS is different for gamma-only and various k-point dependent simulations. First, turn off the k-point algorithm by setting `gamma_only = 0` in `INPUT`, and then set up your own k-points file.

Reviewer: ```suggestion
- **Description**: The name of the k-points file.
- **Note**: In atomic orbitals basis and gamma only calculation, the `KPT` file is unnecessary, because a `KPT` file will be generated automatically. When more than one k-points are required, an explicit `KPT` file is mandatory.

- **Default**: KPT

### pseudo_dir

- **Type**: String

> - **Description**: Specifies the pseudopotential directory.

Reviewer: ```suggestion
- **Description**: The pseudopotential directory.

> - **Default**: / , which refer to the absolute path.

Reviewer: this is confusing, what do you mean by "/"

Author: I have added annotations for absolute paths and relative paths.

Reviewer: No, / always means the root directory of linux system. It can be very confusing if you write it here even if you have explained.it elsewhere.

### orbital_dir

- **Type**: String
- **Description**: Specifies the orbital file directory.

> - **Default**: / , which refer to the absolute path.

Reviewer: confusing in the same way as above

### read_file_dir

- **Type**: String

> - **Description**: Indicates the location of files, such as electron density (`SPIN1_CHG.cube`), required as a starting point. For example, './' implies the file is located in the working directory.

Reviewer: ```suggestion
**Description**: The location of files required for restarting running of ABACUS, such as electron density (`SPIN1_CHG.cube`). For example, './' implies the file is located in the working directory.
```

- **Default**: OUT.$suffix

### wannier_card

- **Type**: String

> - **Description**: Relevant when using ABACUS with Wannier90. Specifies the name of the input file related to Wannier90.

Reviewer: ```suggestion
- **Availability**: Using ABACUS with Wannier90.
- **Description**: The name of the input file related to Wannier90.

## Density of states

> These variables are used to control the calculation of DOS. [Detail introduce](https://github.com/deepmodeling/abacus-develop/blob/develop/docs/advanced/elec_properties/dos.md)

Reviewer: ```suggestion
These variables are used to control the calculation of DOS. [Detailed introduction](https://github.com/deepmodeling/abacus-develop/blob/develop/docs/advanced/elec_properties/dos.md)

### dos_edelta_ev

- **Type**: Real

> - **Description**: Controls the step size in writing Density of States (DOS) (in eV).

Reviewer: ```suggestion
**Description**: The step size for printing Density of States (DOS) (in eV).

- **Default**: 0.01
- **Unit**: eV

### dos_sigma

- **Type**: Real

> - **Description**: Controls the width of the Gaussian factor when obtaining smeared Density of States (DOS) (in eV).

Reviewer: ```suggestion
- **Description**: The width of the Gaussian factor when obtaining smeared Density of States (DOS) (in eV).

- **Default**: 0.07
- **Unit**: eV

### dos_scale

- **Type**: Real
- **Description**: Defines the energy range of DOS output as (emax-emin)*(1+dos_scale), centered at (emax+emin)/2. This parameter will be used when dos_emin and dos_emax are not set.
- **Default**: 0.01

### dos_emin_ev

- **Type**: Real
- **Description**: Specifies the minimal range for Density of States (DOS) (in eV). If set, "dos_scale" will be ignored.
- **Default**: Minimal eigenenergy of $\hat{H}$
- **Unit**: eV

### dos_emax_ev

- **Type**: Real
- **Description**: Specifies the maximal range for Density of States (DOS) (in eV). If set, "dos_scale" will be ignored.
- **Default**: Maximal eigenenergy of $\hat{H}$
- **Unit**: eV

### dos_nche

- **Type**: Integer

> - **Description**: Defines the order of Chebyshev expansions when using Stochastic Density Functional Theory (SDFT) to calculate DOS.

Reviewer: ```suggestion
The order of Chebyshev expansions when using Stochastic Density Functional Theory (SDFT) to calculate DOS.

## Gate field (compensating charge)

> These variables are relevant to gate field (compensating charge) [Detail introduce](https://github.com/deepmodeling/abacus-develop/blob/develop/docs/advanced/scf/advanced.md#compensating-charge)

Reviewer: ```suggestion
These variables are relevant to gate field (compensating charge) [Detailed introduction](https://github.com/deepmodeling/abacus-develop/blob/develop/docs/advanced/scf/advanced.md#compensating-charge)

## Berry phase and wannier90 interface

These variables are used to control berry phase and wannier90 interface parameters. [Detail introduce](https://github.com/deepmodeling/abacus-develop/blob/develop/docs/advanced/interface/Wannier90.md#wannier90)

### berry_phase

- **Type**: Boolean
- **Description**: Controls the calculation of Berry phase
  - true: Calculate Berry phase.
  - false: Do not calculate Berry phase.
- **Default**: false

### gdir

- **Type**: Integer

> - **Description**: Specifies the direction of the polarization in the lattice vector for Berry phase calculation

Reviewer: ```suggestion
- **Description**: The direction of the polarization in the lattice vector for Berry phase calculation.

  - 1: Calculate the polarization in the direction of the lattice vector a_1 defined in the STRU file.
  - 2: Calculate the polarization in the direction of the lattice vector a_2 defined in the STRU file.
  - 3: Calculate the polarization in the direction of the lattice vector a_3 defined in the STRU file.
- **Default**: 3

### towannier90

- **Type**: Integer
- **Description**: Controls the generation of files for the Wannier90 code
  - 1: Generate files for the Wannier90 code.
  - 0: Do not generate files for the Wannier90 code.
- **Default**: 0

### nnkpfile

- **Type**: String
- **Description**: The file name generated when running "wannier90 -pp ..." command.
- **Default**: seedname.nnkp

### wannier_spin

- **Type**: String

> - **Description**: Specifies the spin direction for the Wannier function calculation when nspin is set to 2.

Reviewer: ```suggestion
- **Description**: The spin direction for the Wannier function calculation when nspin is set to 2.
